analyzers/src/SonarAnalyzer.Common/Rules/Utilities/SymbolReferenceAnalyzerBase.cs

```diff
@@ -33,40 +34,24 @@ public abstract class SymbolReferenceAnalyzerBase<TSyntaxKind> : UtilityAnalyzer
         private const string Title = "Symbol reference calculator";
         private const int TokenCountThreshold = 40_000;
 
-        private readonly ISet<SymbolKind> declarationKinds = new HashSet<SymbolKind>
-        {
-            SymbolKind.Event,
-            SymbolKind.Field,
-            SymbolKind.Local,
-            SymbolKind.Method,
-            SymbolKind.NamedType,
-            SymbolKind.Parameter,
-            SymbolKind.Property,
-            SymbolKind.TypeParameter
-        };
+        protected sealed override string FileName => "symrefs.pb";
 
         protected abstract SyntaxNode GetBindableParent(SyntaxToken token);
 
-        protected sealed override string FileName => "symrefs.pb";
+        protected abstract ReferenceInfo[] CreateDeclarationReferenceInfo(SyntaxNode node, SemanticModel model);
```

Comment:
If you make this `IEnumerable<ReferenceInfo>`, only one place would need a `.ToArray()` call. The place where the method is invoked, and all other `ToArray` calls in the C# and VBNet classes could be removed.
```suggestion
        protected abstract IEnumerable<ReferenceInfo> CreateDeclarationReferenceInfo(SyntaxNode node, SemanticModel model);
```

Reply:
Which was the old behavior. Could you explain why this change was made?

Reply:
For most of the cases, we already return arrays. See `CreateDeclarationReferenceInfo`. If we return `IEnumerable<ReferenceInfo>` we will have to create copies of those arrays.